terraform/lib/dependabot/terraform/file_parser.rb

```diff
@@ -54,6 +57,28 @@ def parse
 
       private
 
+      def build_provider_dependency_from(file, details = {})
+        source_address = details.fetch("source")
+        version = details["version"]&.strip
+        hostname, namespace, name = provider_source_from(source_address)
+
+        Dependency.new(
+          name: "#{namespace}/#{name}",
+          version: version, # resolved version should come from `.terraform.lock.hcl`.
+          package_manager: "terraform",
+          requirements: [
+            requirement: version,
+            groups: [],
+            file: file.name,
+            source: {
+              type: "provider",
+              registry_hostname: hostname,
+              module_identifier: "#{namespace}/#{name}"
+            }
+          ]
+        )
+      end
+
       def build_terraform_dependency(file, name, details, provider)
```

Comment:
Can the `provider` arg be removed here? Looks like it might always be `false` now

Reply:
I think it can. Would it be okay for me to do that in this PR or would you prefer that to be a separate one?